test/jdk/jdk/jfr/tool/TestMetadata.java

```diff
@@ -74,5 +82,23 @@ public static void main(String[] args) throws Throwable {
             }
             lineNumber++;
         }
+
+        output = ExecuteHelper.jfr("metadata");
+        List<String> eventNames = new ArrayList<>();
+        List<String> lines = output.asLines();
+        for (String line : lines) {
+            if (line.startsWith("@Name(\"")) {
+                eventNames.add(line.substring(7, line.indexOf("\"", 7)));
+            }
+        }
+        List<Type> eventTypes = TypeLibrary.getInstance().getTypes();
```

Comment:
I think it would be better to use public API (FlightRecorder.getEventTypes()) instead of calling internals.

Reply:
Changed.